macros/plugins/spark/get_external_build_plan.sql

```diff
@@ -19,6 +19,13 @@
         {% set build_plan = build_plan + dbt_external_tables.refresh_external_table(source_node) %}
     {% endif %}
 
+    {% set recover_partitions = spark__recover_partitions(source_node) %}
+    {% if recover_partitions|length > 0 %}
```

Comment:
@jtcohen6 : outside this PR, it might be neat to skip sql statements that are empty (after `.strip()`)

Reply:
Good call. For other macros, I handle this by returning an empty list ([e.g. here](https://github.com/dbt-labs/dbt-external-tables/blob/c58cc5ec10831cf3158ee54c5226304d3554bdfa/macros/plugins/snowflake/refresh_external_table.sql#L11-L25)), which has the effect of adding nothing to the `build_plan`